### PR TITLE
remove `cdms2`

### LINKED ISSUE #8419: Instructions to create a development environment for M1/M2/osx-arm64
### What is your issue?

The current instructions to [create a development environment](https://docs.xarray.dev/en/stable/contributing.html#creating-a-python-environment) for MacOS don't work for an M1/M2 Mac.

The conda environment solve fails on cdsm2 presumably because it's not available for osx-arm64.  

I tried looking for related discussion and/or issues with recommendations, but didn't see any.

### PR DESCRIPTION
**Pull request by keewis**
`cdms2` is going to be abandoned at the end of this year, the recommended replacement is `xcdat`. We have deprecated our conversion functions in 2023.06.0 (release on June 23) which was less than 6 months ago, but given that `cdms2` is already not installable on some architectures it makes sense to remove earlier.

This also appears to allow us to remove the special `python=3.11` environment files.

- [x] Follow-up to #7876, closes #8419
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by keewis** [comment]
there's some mentions of `cdms2` / CDAT in the faq, so this will have to be updated before merging

File: doc/getting-started-guide/faq.rst [hunk 1]
```diff
@@ -168,18 +168,11 @@ integration with Cartopy_.
...
-
 We think the design decisions we have made for xarray (namely, basing it on
 pandas) make it a faster and more flexible data analysis tool. That said, Iris
-and CDAT have some great domain specific functionality, and xarray includes
-methods for converting back and forth between xarray and these libraries. See
-:py:meth:`~xarray.DataArray.to_iris` and :py:meth:`~xarray.DataArray.to_cdms2`
-for more details.
+has some great domain specific functionality, and xarray includes
+methods for converting back and forth between xarray and Iris. See
+:py:meth:`~xarray.DataArray.to_iris` for more details.
```
Comment (keewis) [inline doc/getting-started-guide/faq.rst:175]: with this `cdms2` is gone from the FAQ, but I'm not at all sure if this was the best way to do this. Could someone with more knowledge in this area double-check, please?

Comment (dcherian) [reply doc/getting-started-guide/faq.rst:175]: LGTM!

**Comment by keewis** [comment]
@dcherian, there's some `groupby` failures that are not visible on `main` (or maybe they will appear soon?). Can you make sense of these?

**Comment by dcherian** [comment]
Should be fixed after a numbagg release.

**Pull request merged**